articles/architecture-scenarios/implementation/b2b/b2b-logout.md

```diff
@@ -14,10 +14,45 @@ useCase:
 
 <%= include('../../_includes/_logout/_introduction.md', { platform: 'b2b' }) %>
 
+## Single Logout
+
+If you are doing [Federated Logout](#federated-logout) you will likely also want to do Single Logout (SLO).
+
+::: warning
+SLO can add complexity to your system, so you need to ensure that you really need it before adding the extra development and maintenance time to your system.
+:::
+
+There are two main approaches you can take to Single Logout:
+
+### Short-lived tokens
+
+This is by far the simplest approach to Single Logout.  Each application enforces a short time that a user can use the system (5-10 minutes) and then on each action by that user if the time has expired, it will either redirect to Auth0 (for regular web apps), or use [Silent Authentication](https://auth0.com/docs/api-auth/tutorials/silent-authentication) for client side tokens.  This will ensure that eventually (within a minute or two) all other applications will fail to get their refreshed token because the SSO session has been removed through logout.
+
+::: panel Best Practice
+You want to avoid making too many calls to your Auth0 tenant to avoid rate limiting and poor performance.  A best practice is to only request new tokens if tokens have expired and a user takes an action.  This will avoid applications that are simply open, but not in use, from continually polling for new tokens.
+:::
+
+::: warning
+If you are automatically forwarding the user directly to their own IdP as part of an enterprise connection using the connection parameter, this can break this technique unless you are also doing [Federated Logout](#federated-logout)
+:::
+
+### Build a logout service
+
+Another technique you can do is build a logout service that can track and destroy application sessions.  Each application will notify the logout service when it creates and removes a session.  This logout service will either have direct access to all application's server side sessions and destroy them directly, or it will have the ability to make a back-channel call to each application to tell the application that it must remove its session.  This technique can be effective because it has low-latency between when a user calls logout and when they are logged out of all applications, but it can add complexity and development time for its creation.  It will also require some way to ensure that new applications added to the system are added to this service.
```

Comment:
```suggestion
Another technique you can use is to build a logout service that can track and destroy application sessions. Each application would notify the logout service when it creates and removes a session. The (logout) service would either have direct access to all application's server side sessions and destroy them directly, or it will have the ability to make a back-channel call to each application to tell the application that it must remove its session. 

This technique can be quite effective as there is low-latency between when a user calls logout, and when they are then logged out of all applications. However it can add complexity and also additional development time for implementation. It will also require some way to ensure that new applications added to the system are added to this service.
```